### ISSUE TITLE
Social Login

### ISSUE THREAD

**Issue by rudy1976s** [CakeDC/users#933]
I have problems with social login with Twitter and Linkedin.
My application has different plugins installed then I decided to test a fresh installation, to be sure that problems would occur the same way and unfortunately it does.
I installed a fresh cakephp 4.2.2 and CakeDC plugin 9.1
I followed documentation ( enabled Social login, setup credentials etc, installed dependencies ) setup linkedin,Twitter apps on their side.
Both social login does not work right after passing the social auth ( both on twitter and linkedin ).
There is a redirect to a "/users/social-email"  : I tried to add the email but nothing happens : the page redirect to itself without errors.

Google and Facebook worked like a charm.

I also noticed that instagram login is no longer supported ( they advice to use facebook login instead )

Rudy

**Comment by tgoeminne** [comment]
I also found this problem with linkedin it forwards the user to social email page, why does that happen?

**Comment by rochamarcelo** [comment]
Linkedin is not returning user e-mail. Have you checked this?

**Comment by steinkel** [comment]
Confirmed for LinkedIn. I've just did a quick test with a new application and minimal plugin setup. Overriding the LinkedIn provider to bypass the initial errors about not authorized scope `r_liteprofile` that's deprecated according to their documentation https://learn.microsoft.com/en-us/linkedin/consumer/integrations/self-serve/sign-in-with-linkedin.

```
<?php
namespace App\Oauth2\Client\Provider;
use League\OAuth2\Client\Provider\LinkedIn as BaseLinkedIn;
class LinkedIn extends BaseLinkedIn
{
    public function getDefaultScopes(): array
    {
        return ['email', 'openid', 'profile'];
    }
}

```

Even in that case I got to a `Not enough permissions to access: GET /me` reported by other users in several places.

They are pushing for OpenID-Connect as the default connection moving forward. They don't even offer options to configure the old OAuth 2.0 login in their application "Products" tab.

So, we would be marking the LinkedIn (using OAuth 2.0) as deprecated/not working and explore OpenID-Connect to provide an adapter in the future.
